### PR TITLE
Remove singleton instances from debugger code 

### PR DESCRIPTION
**Pull request by lankavitharana**
Fix https://github.com/ballerinalang/ballerina/issues/4130

### CODE REVIEW

File: modules/ballerina-core/src/main/java/org/ballerinalang/util/debugger/VMDebugClientHandler.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import org.ballerinalang.util.debugger.info.BreakPointInfo;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * {@code VMDebugSession} The Debug Session class will be used to hold context for each client.
+ * Each client will get its own instance of debug session.
+ *
+ * @since 0.88
```
Comment (anupama-pathirage) [inline modules/ballerina-core/src/main/java/org/ballerinalang/util/debugger/VMDebugClientHandler.java]: Incorrect version?

File: modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/DebuggerExecutor.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import org.ballerinalang.util.debugger.dto.BreakPointDTO;
+import org.ballerinalang.util.exceptions.BallerinaException;
+import org.ballerinalang.util.program.BLangFunctions;
+
+import java.util.List;
+
+/**
+ * {@link DebuggerExecutor} represents executor class which runs the main program when debugging.
+ *
+ * @since 0.88
```
Comment (anupama-pathirage) [inline modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/DebuggerExecutor.java]: Incorrect version?

File: modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/DebuggerExecutor.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        ControlStackNew controlStackNew = bContext.getControlStackNew();
+        controlStackNew.pushFrame(stackFrame);
+        bContext.startTrackWorker();
+        bContext.setStartIP(defaultWorkerInfo.getCodeAttributeInfo().getCodeAddrs());
+
+        BLangVM bLangVM = new BLangVM(programFile);
+        bLangVM.run(bContext);
+        bContext.await();
+        debugManager.notifyExit();
+
```
Comment (anupama-pathirage) [inline modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/DebuggerExecutor.java]: Remove extra new line here

File: modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/TestDebugClientHandler.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        }
+
+        @Override
+        public DebugContext getContext(String threadId) {
+            return this.contextMap.get(threadId);
+        }
+
+        @Override
+        public void updateAllDebugContexts(DebugCommand debugCommand) {
+
```
Comment (anupama-pathirage) [inline modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/TestDebugClientHandler.java]: Remove extra new line here. Check other methods in this class too

File: modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/VMDebuggerUtil.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+import org.ballerinalang.util.debugger.dto.BreakPointDTO;
+import org.testng.Assert;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * Test Util class to test debug scenarios.
+ *
+ * @since 0.90
```
Comment (anupama-pathirage) [inline modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/VMDebuggerUtil.java]: Incorrect version?

File: modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/VMDebuggerUtil.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        debugManager.setDebugEnabled(true);
+
+        String[] args = {"Hello", "World"};
+
+        DebuggerExecutor executor = new DebuggerExecutor(result, args, clientHandler, debugServer,
+                new ArrayList<>(Arrays.asList(breakPoints)));
+        (new Thread(executor)).start();
+        return debugManager;
+    }
+
```
Comment (anupama-pathirage) [inline modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/VMDebuggerUtil.java]: Remove extra new line here

File: modules/ballerina-core/src/main/java/org/ballerinalang/util/debugger/VMDebugClientHandler.java (hunk 1 above)
Comment (lankavitharana) [reply modules/ballerina-core/src/main/java/org/ballerinalang/util/debugger/VMDebugClientHandler.java]: no this was a class rename

File: modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/DebuggerExecutor.java (hunk 2 above)
Comment (lankavitharana) [reply modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/DebuggerExecutor.java]: class rename

File: modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/VMDebuggerUtil.java (hunk 5 above)
Comment (lankavitharana) [reply modules/ballerina-test/src/test/java/org/ballerinalang/test/utils/debug/VMDebuggerUtil.java]: this class is moved(not a new one)

**Pull request merged**